--- dlls/gdi32/freetype.c.orig	2008-10-17 23:45:25.000000000 +0800
+++ dlls/gdi32/freetype.c	2009-10-31 14:45:15.394598327 +0800
@@ -3313,16 +3313,33 @@
     if(!strcmpiW(lf.lfFaceName, SymbolW))
         lf.lfCharSet = SYMBOL_CHARSET;
 
-    if(!TranslateCharsetInfo((DWORD*)(INT_PTR)lf.lfCharSet, &csi, TCI_SRCCHARSET)) {
+    if( lf.lfCharSet == DEFAULT_CHARSET || lf.lfCharSet == ANSI_CHARSET)
+    {
+        int codepage = 936; /* I'm Chinese. Let me do something selfish. */
+        char *env_wdc = getenv("WINE_DEFAULT_CODEPAGE");
+        if (env_wdc != NULL)
+        {
+            codepage = atoi(env_wdc);
+        }
+        FIXME("Dirty hack. Default and ansi charset is translated to getenv(\"WINE_DEFAULT_CODEPAGE\")\n");
+        if(!TranslateCharsetInfo((DWORD*)(INT_PTR)codepage, &csi, TCI_SRCCODEPAGE)) 
+        {
+            FIXME("OMG. Even this dirty hack doesn't work.  Let's PRAY.\n");
+            csi.fs.fsCsb[0] = 0;
+        }
+        lf.lfCharSet = csi.ciCharset;
+    }
+    else if(!TranslateCharsetInfo((DWORD*)(INT_PTR)lf.lfCharSet, &csi, TCI_SRCCHARSET)) 
+    {
         switch(lf.lfCharSet) {
-	case DEFAULT_CHARSET:
-	    csi.fs.fsCsb[0] = 0;
-	    break;
-	default:
-	    FIXME("Untranslated charset %d\n", lf.lfCharSet);
-	    csi.fs.fsCsb[0] = 0;
-	    break;
-	}
+            case DEFAULT_CHARSET:
+                csi.fs.fsCsb[0] = 0;
+                break;
+            default:
+                FIXME("Untranslated charset %d\n", lf.lfCharSet);
+                csi.fs.fsCsb[0] = 0;
+                break;
+        }
     }
 
     family = NULL;
