--- 915resolution/915resolution.c
+++ 915resolution/915resolution.c
@@ -885,16 +885,15 @@
 
 
 
-
+static grub_command_t cmd;
 
 GRUB_MOD_INIT(915resolution)
 {
-  (void)mod;			/* To stop warning. */
-  grub_register_command ("915resolution", grub_cmd_915resolution, GRUB_COMMAND_FLAG_BOTH,
-			 "915resolution", "Intel VBE editor", 0);
+  cmd = grub_register_command ("915resolution", grub_cmd_915resolution,
+			 0, "Intel VBE editor");
 }
 
 GRUB_MOD_FINI(915resolution)
 {
-  grub_unregister_command ("915resolution");
+  grub_unregister_command (cmd);
 }
