diff --git a/extra/PythonSerializer.py b/extra/PythonSerializer.py
index 47d8c83..cc9df05 100644
--- a/extra/PythonSerializer.py
+++ b/extra/PythonSerializer.py
@@ -17,9 +17,9 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from rdflib.syntax.serializers.RecursiveSerializer import RecursiveSerializer
+from rdflib.plugins.serializers.turtle import RecursiveSerializer
 from rdflib import RDF, RDFS
-from rdflib.Namespace import Namespace
+from rdflib.namespace import Namespace
 
 NIENS = Namespace("http://www.semanticdesktop.org/ontologies/2007/01/19/nie#")
 
diff --git a/extra/rdfxml2py b/extra/rdfxml2py
index bd8dd23..19de9e8 100755
--- a/extra/rdfxml2py
+++ b/extra/rdfxml2py
@@ -9,16 +9,17 @@
 import rdflib
 
 from rdflib.plugin import register
-from rdflib.syntax import serializer, serializers
+from rdflib.plugins import serializers
+from rdflib import serializer
 
-register('python', serializers.Serializer,
+register('python', serializer.Serializer,
          'PythonSerializer', 'PythonSerializer')
 
 def parse(trig_stream):
 	"""
 	Return a list of triples representing the ontology
 	"""
-	trig_in = rdflib.FileInputSource(trig_stream)
+	trig_in = rdflib.parser.FileInputSource(trig_stream)
 	ontology = rdflib.ConjunctiveGraph()
 	ontology.parse(trig_in)
 	pycode = ontology.serialize(format="python")
