--- a/configure.ac	2010-10-27 02:52:14.272000028 +0800
+++ b/configure.ac	2010-10-27 03:09:42.572000028 +0800
@@ -47,8 +47,8 @@
                              [Enable libnotify support (used to show message notifications) @<:@default=yes@:>@]),,
               enable_libnotify="yes")
 
-AC_ARG_ENABLE(gstreamer,
-              AS_HELP_STRING([--enable-gstreamer],
+AC_ARG_ENABLE(gst,
+              AS_HELP_STRING([--enable-gst],
                              [Enable gstreamer support (used to play alert sounds) @<:@default=yes@:>@]),,
               enable_gst="yes")
 
@@ -62,7 +62,7 @@
 			AC_DEFINE(USE_GTK, 1, [Use GTK])
 			AC_SUBST(GTK_CFLAGS)
 			AC_SUBST(GTK_LIBS)
-		] ,	
+		] ,
 			)
 # checks for gthread-2.0
 PKG_CHECK_MODULES(GTHREAD, [gthread-2.0], [
@@ -139,7 +139,7 @@
 			echo ""
 			echo "configure: error: "
 			echo "gstreamer development headers not found."
-			echo "Use --disable-gstreamer if you do not need it."
+			echo "Use --disable-gst if you do not need it."
 			echo "If you're using debian or ubuntu,install it by "
 			echo "\"sudo apt-get install libgstreamer0.10-dev\""
 			echo ""
