--- /dev/null2007-09-19 15:17:56.000000000 +0100
+++ fontconfig-2.7.2.new/conf.d/10-antialias.conf	2009-08-18 20:34:36.214531075 +0800
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use the Antialiasing --> 
+  <match target="font">
+    <edit name="antialias" mode="assign"><bool>true</bool></edit>
+  </match>
+</fontconfig>
--- /dev/null20072007-09-19 15:17:56.000000000 +0100
+++ fontconfig-2.7.2.new/conf.d/10-hinting.conf	2009-08-18 20:34:36.214531075 +0800
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Hinting --> 
+  <match target="font">
+    <edit name="hinting" mode="assign"><bool>true</bool></edit>
+  </match>
+</fontconfig>
--- /dev/null200720072007-09-19 15:17:56.000000000 +0100
+++ fontconfig-2.7.2.new/conf.d/10-hinting-full.conf	2009-08-18 20:34:36.215510180 +0800
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Full Hinting --> 
+  <match target="font">
+    <edit name="hintstyle" mode="assign"><const>hintfull</const></edit>
+  </match>
+</fontconfig>
--- /dev/null2007200720072007-09-19 15:17:56.000000000 +0100
+++ fontconfig-2.7.2.new/conf.d/10-hinting-medium.conf	2009-08-18 20:34:36.215510180 +0800
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Medium Hinting --> 
+  <match target="font">
+    <edit name="hintstyle" mode="assign"><const>hintmedium</const></edit>
+  </match>
+</fontconfig>
--- /dev/null20072007200720072007-09-19 15:17:56.000000000 +0100
+++ fontconfig-2.7.2.new/conf.d/10-hinting-slight.conf	2009-08-18 20:34:36.216570301 +0800
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Slight Hinting --> 
+  <match target="font">
+    <edit name="hintstyle" mode="assign"><const>hintslight</const></edit>
+  </match>
+</fontconfig>
